ui/lib/ldap/addon/components/page/role/details.ts

```diff
@@ -0,0 +1,30 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import errorMessage from 'vault/utils/error-message';
+
+import type LdapRoleModel from 'vault/models/ldap/role';
+import { Breadcrumb } from 'vault/vault/app-types';
+import type FlashMessageService from 'vault/services/flash-messages';
+import type RouterService from '@ember/routing/router-service';
+
+interface Args {
+  model: LdapRoleModel;
+  breadcrumbs: Array<Breadcrumb>;
+}
+
+export default class LdapRoleDetailsPageComponent extends Component<Args> {
+  @service declare readonly flashMessages: FlashMessageService;
+  @service declare readonly router: RouterService;
+
+  @action
+  async delete() {
+    try {
+      await this.args.model.destroyRecord();
+      this.router.transitionTo('vault.cluster.secrets.backend.ldap.roles');
```

Comment:
Should we add a success flash here?

Reply:
It looks to be pretty inconsistent across the app but I think it's beneficial to add it in to confirm that the operation was successful.